spec/controllers/api/public/v1/providers/courses_controller_spec.rb

```diff
@@ -0,0 +1,57 @@
+require "rails_helper"
+
+RSpec.describe API::Public::V1::Providers::CoursesController do
+  let(:provider) { create(:provider) }
+  let(:recruitment_cycle) { provider.recruitment_cycle }
+
+  describe "#index" do
+    context "when there are no courses" do
+      it "returns empty array of data" do
+        get :index, params: {
+          recruitment_cycle_year: recruitment_cycle.year,
+          provider_code: provider.provider_code,
+        }
+        expect(JSON.parse(response.body)["data"]).to eql([])
+      end
+    end
+
+    context "when there are courses" do
+      before do
+        create(:course, provider: provider)
+        create(:course, provider: provider)
+      end
+
+      it "returns correct number of courses" do
+        get :index, params: { recruitment_cycle_year: "2020", provider_code: provider.provider_code }
```

Comment:
same for this eg https://github.com/DFE-Digital/teacher-training-api/pull/1517/files#r470030625